libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1161,25 +1206,53 @@ void WaveClip::ShiftBy(double delta) noexcept
 
 // Bug 2288 allowed overlapping clips.
 // This was a classic fencepost error.
-// We are within the clip if start < t <= end.
+// We are within the clip if start <= t < end.
 // Note that BeforeClip and AfterClip must be consistent
 // with this definition.
 bool WaveClip::WithinPlayRegion(double t) const
 {
-    auto ts = TimeToSamples(t);
-    return ts > GetPlayStartSample() && ts < GetPlayEndSample() + GetAppendBufferLen();
+   return GetPlayStartTime() <= t && t < GetPlayEndTime();
+}
+
+bool WaveClip::EntirelyWithinPlayRegion(double t0, double t1) const
+{
+   assert(t0 <= t1);
```

Comment:
What you assert at the top of this and the following new functions should be mentioned in the header file comments as `@pre`

Reply:
I added those `@pre` later and squashed them in the relevant commit.